internal/kubelet/client.go

```diff
@@ -165,9 +166,27 @@ func (p *saClientProvider) BuildClient() (Client, error) {
 	}
 	tr := defaultTransport()
 	tr.TLSClientConfig = &tls.Config{
-		RootCAs: rootCAs,
+		RootCAs:            rootCAs,
+		InsecureSkipVerify: true,
+	}
+	endpoint, err := buildEndpoint(p.endpoint, true, p.logger)
+	if err != nil {
+		return nil, err
+	}
+	httpClient := http.Client{
+		Transport: tr,
+	}
+	httpClient.Transport, err = transport.NewBearerAuthWithRefreshRoundTripper(string(tok), p.tokenPath, tr)
+	if err != nil {
+		return nil, err
 	}
-	return defaultTLSClient(p.endpoint, true, rootCAs, nil, tok, p.logger)
+
+	return &clientImpl{
+		baseURL:    endpoint,
+		httpClient: httpClient,
+		tok:        nil,
+		logger:     p.logger,
+	}, nil
```

Comment:
Seems like a redundant assignment of `httpClient.Transport` if we override it in the next step
```suggestion
	tr, err = transport.NewBearerAuthWithRefreshRoundTripper(string(tok), p.tokenPath, tr)
	if err != nil {
		return nil, err
	}
	return &clientImpl{
		baseURL:    endpoint,
		httpClient: http.Client{Transport: tr},
		tok:        nil,
		logger:     p.logger,
	}, nil
```

Reply:
Yes, i can cleanup defaultTLSClient when looking at the other config issue. Re: redundant assignment, does not work due to multiple assignments to tr in that call.

Reply:
> Re: redundant assignment, does not work due to multiple assignments to tr in that call

Not sure I understand. Doesn't my suggestion work? all the `tr` assignments are of the same type, right?